component/discovery/file/file_test.go

```diff
@@ -30,7 +30,8 @@ func TestFile(t *testing.T) {
 	})
 	c := createComponent(t, dir, []string{path.Join(dir, "*.txt")}, nil)
 	ct := context.Background()
-	ct, _ = context.WithTimeout(ct, 5*time.Second)
+	ct, ccl := context.WithTimeout(ct, 5*time.Second)
+	defer ccl()
```

Comment:
Here and below, the linter was not happy to discard the cancel function, so I made sure we run it. I suspect that the linting highlighted it now because it was a new rule added since this file was last time modified?

Reply:
Should rerun the linter from scratch, are you running `make lint` or local linting? Also our make lint sometimes lags.

Reply:
Yeah, I used `make lint` - it is now happy locally and in CI. But before this change I was getting a lot of:
```
component/discovery/file/file_test.go:33:6: lostcancel: the cancel function returned by context.WithTimeout should be called, not discarded, to avoid a context leak (govet)
	ct, _ = context.WithTimeout(ct, 5*time.Second)
	    ^
```